operators/multiclusterobservability/controllers/placementrule/hub_info_secret.go

```diff
@@ -14,14 +14,15 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	mcov1beta2 "github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/api/v1beta2"
 	"github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/pkg/config"
 	operatorconfig "github.com/stolostron/multicluster-observability-operator/operators/pkg/config"
 )
 
 // generateHubInfoSecret generates the secret that contains hubInfo.
 // this function should only called when the watched resources are created/updated.
 func generateHubInfoSecret(client client.Client, obsNamespace string,
-	namespace string, ingressCtlCrdExists bool) (*corev1.Secret, error) {
+	namespace string, ingressCtlCrdExists bool, mco *mcov1beta2.MultiClusterObservability) (*corev1.Secret, error) {
```

Comment:
Why don't we follow the same logic as for the platform alerting disablement?

I.e instead of passing in the whole MCO, we create a variable in the config, and use that instead of having to pass the whole MCO object here.

Reply:
Maybe this new way is a bit simpler (requires less logic maybe), but would be best to handle them the same.

Maybe we could create a task to refactor this code so it is the same for both the uwm and platform alerting in 2.15?